Assets/Scripts/Features/UniversalGroundChecker.cs

```diff
@@ -0,0 +1,118 @@
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Baz_geluk9.Feature_vault.Feature
+{
+    /// <summary>
+    /// Universal ground checker supporting 2D and 3D environments with line and sphere casts.
+    /// Supports both offset and transform-based ground checking.
+    /// </summary>
+    public sealed class UniversalGroundChecker : MonoBehaviour
+    {
+        [Header("Usage")]
+        [SerializeField, Tooltip("True for 3D settings.\nFalse for 2D settings.")] private bool is3D = true;
```

Comment:
Why not split 2D and 3D in their own seperate classes with either a base class or interface?

Reply:
I did not apply the feedback directly. The _origin3D & _origin2D now aren't calculated both in the fixed update.